packages/nextra-theme-docs/src/components/flexsearch.tsx

```diff
@@ -271,8 +511,46 @@ export function Flexsearch({
       }
       setLoading(false)
     }
-    const newResults = getResults(value, locale)
-    if (newResults) setResults(newResults)
+
+    setResults(
+      getResults(value, locale).flatMap(page => {
+        return page.sections.map((section, i) => {
+          const route = section.anchor
+            ? `${page.route}#${section.anchor}`
+            : page.route
+
+          return {
+            id: route,
+            route,
+            prefix: i == 0 && (
+              <div
+                className={cn(
+                  '_mx-2.5 _mb-2 _mt-6 _select-none _border-b _border-black/10 _px-2.5 _pb-1.5 _text-xs _font-semibold _uppercase _text-gray-500 first:_mt-0 dark:_border-white/20 dark:_text-gray-300',
+                  'contrast-more:_border-gray-600 contrast-more:_text-gray-900 contrast-more:dark:_border-gray-50 contrast-more:dark:_text-gray-50'
+                )}
+              >
+                <Content parts={page.title} />
+              </div>
+            ),
+            children: (
+              <>
+                <div className="_text-base _font-semibold _leading-5">
+                  <Content
+                    parts={section.title.length ? section.title : page.title}
```

Comment:
```suggestion
                    parts={section.title || page.title}
```

Reply:
These are now lists, so this pattern doesn't work anymore.